src/utils/sampler.py

```diff
@@ -0,0 +1,143 @@
+"""
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import ee
+
+
+class SampleExporter:
+    """Manage export operations, takes images to extract patches."""
+
+    def __init__(self, task_manager, num_samples, num_shards, kernel, scale,
+                 bucket, directory):
+        """Initialize the SampleExport object.
+
+        Parameters
+        ----------
+        task_manager : TaskManager
+            TaskManager object to submit the tasks to
+        num_samples : int
+            Number of samples to extract from each image.
+        num_shards : int
+            Number of shards used to parition the samples.
+        kernel : ee.Kernel
+            kernel used for neighborhoodtoArray
+        scale : int
+            scale parameter for sampling and export (in meters/pixel)
+        bucket : str
+            Google Cloud Storage Bucke name
+        directory : str
+            Folder to extract the TFRecords to
+
+        Returns
+        -------
+        None.
+
+        """
+        # Task parameters
+        self.task_manager = task_manager
+        self.num_samples = num_samples
+        self.num_shards = num_shards
+
+        # Patches paramters
+        self.kernel = kernel
+        self.scale = scale
+
+        # Storage location
+        self.bucket = bucket
+        self.directory = directory
+
+    def export_patches(self, image, bands, patch_bands, export_id):
+        """Export patch and scalar bands and submit tasks to the task manager.
+
+        Parameters
+        ----------
+        image : ee.Image
+            Image from which to extract patches. It should have a 'valid' band.
```

Comment:
Nit: Will it always have a valid band? If so, it would be clearer to reword as, "Assumed to have a 'valid' band."